### DISCUSSION TITLE
Add support for creating and editing beatmaps of custom rulesets

### DISCUSSION THREAD

**Discussion by RichardKanshen** [ppy/osu#14636] [Q&A] [answered]
Almost every ruleset master has inserted the similar text of the following one in README files:

> ## TODO : * Better player sprite * Boss sprite during kiai sections * Hammers and other minion variants * Custom avatars/abilities via mods? * **Editor (waiting on updates in lazer)** * (Stated from Rush ruleset master (README.md))

But, because of osu!filesystem, a custom ruleset editor might not be possible. This was caused because of a little piece of code in osu!beatmaps file (.osu file):
`Mode: 0`
Changing number to ruleset filename might fix this issue, like this:
`Mode: osu.Game.Rulesets.Osu.dll`
But since there might be thousands of beatmaps out there, osu!devs would need hours to create editor code for this, and then hundreds of hours just to recode beatmaps. 

This is my thought. Yours?

- Richard
:)

**Comment by bdach** [comment, marked as answer]
this would not be added to the legacy `.osu` format. this will be supported going forward with a brand new format.

that will come in due course. for now the focus is finishing the actual baseline game before more work on rulesets.

**Reply by TheFearlessDeath500** [reply]
there is no new format. it's been 5 years.

**Reply by bdach** [reply]
i wonder why. maybe that it's because it's been that many years and lazer *still* isn't even the most popular osu! client and we're still working on making it that? no. can't be. we're just dragging our feet as usual aren't we.
